api/clients/v2/verification/errors.go

```diff
@@ -0,0 +1,42 @@
+package verification
+
+import "github.com/Layr-Labs/eigenda/api/clients/v2/coretypes"
+
+// CertVerifierInputError represents a 4xx-like error (invalid input, misuse, unsupported, etc.)
+type CertVerifierInputError struct {
+	Msg string
```

Comment:
is there any merit in capturing the status code enum in this type?

Reply:
wdym by status code? like 400 vs 404 vs 403 etc? I don't think we should return http code from library function calls personally. Perhaps I shouldnt have used 4xx and just sticked to "wrong input error type"?
In any case, I do agree we could go a lot more fine grained than 4xx/5xx/certInvalid coarse grained errors that I defined here, but for now the only thing we really need to know in proxy is whether a cert is (in)valid, so I (at this point) didn't see value in going more fine grained. If ever we do want more fine grained detail to type switch on and perform different behaviors inside proxy (for example retry 5xx errors?), then we should! Not sure if there is any value for 4xx errors though...? Haven't thought about it super deeply yet

Reply:
@bxue-l2 you're looking at the wrong error. This error is when an input is wrong. Like we weren't even able to call the CertVerifier contract or check rbn recency because what was sent is not even a cert for example. This error represents a bug, not an invalid cert.

You want to look at `CertVerificationFailedError`, which does include the StatusCode returned by the contract. I regret naming it that way.. probably confused you guys. How about we rename it to `InvalidCertError` or something of that sort? I suggest we do this after we move proxy to monorepo (can also move RBN check to client once we do that)